app/assets/javascripts/app/discussions/discussion.js

```diff
@@ -5,12 +5,14 @@
     .module('council.discussions')
     .controller('DiscussionCtrl', DiscussionCtrl);
 
-  function DiscussionCtrl(Discussion, discussionId, _) {
+  function DiscussionCtrl(Discussion, Comment, discussionId) {
     var ctrl = this;
 
     ctrl.pageReady = false;
-    ctrl.toggleNewComment = toggleNewComment;
-    ctrl.toogleDiscussionState = toogleDiscussionState;
+    ctrl.openEditForm = openEditForm;
+    ctrl.toggleCommentForm = toggleCommentForm;
+
+    ctrl.comment = Comment.createInstance();
```

Comment:
This instance creation should only exist in one place. Make this happen in the form controller alone